server/src/test/java/org/elasticsearch/index/mapper/ContentPathPathAsTextTests.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.Collections;
+import java.util.List;
+
+public class ContentPathPathAsTextTests extends ESTestCase {
```

Comment:
Great tests, thanks for adding those, it makes code changes like these so much easier to check. 
Unfortunately there was another test class for ContentPath added with #98332 (ContentPathTests).
I would like to keep that one as the main test class. Would you mind to merge the test cases that are not covered to that unit test class? I think there might be some duplication now.